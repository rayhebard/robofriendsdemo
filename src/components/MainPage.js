import React, { Component } from 'react';

import CardList from '../components/CardList';
import SearchBox from '../components/Searchbox';
import Scroll from '../components/Scroll';
import ErrorBoundry from '../components/ErrorBoundry';
import Header from '../components/Header';
//import { robots } from '../robots';
import './MainPage.css'

class MainPage extends Component{

   componentDidMount(){
     this.props.onRequestRobots();
     console.log(process.env.REACT_APP_SAY_HI);
   }

   filterRobots = () => {
     return this.props.robots.filter(robot => {
       return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
     })
   }

  render(){
    const{ onSearchChange, isPending} = this.props;

    return isPending ?
      <h1>Loading</h1> :
      (
        <div className='tc'>
          <Header />
          <SearchBox searchChange={onSearchChange}/>
          <Scroll>
            <ErrorBoundry>
              <CardList robots={this.filterRobots()}/>
            </ErrorBoundry>
          </Scroll>
        </div>
      );
  }
}

export default MainPage
