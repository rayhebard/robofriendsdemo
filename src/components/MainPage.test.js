import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import { shallow } from 'enzyme';

import React from 'react'
import MainPage from './MainPage';

let wrapper;
beforeEach( () => {
  const mockProps = {
    onRequestRobots:jest.fn(),
    robots:[{
      id:3,
      name: 'John',
      email: 'john@gmail.com'
    } ],
    searchField:'',
    isPending: false
  }
  wrapper = shallow( <MainPage {...mockProps}/>)
})

it('renders MainPage without crashing', () => {
  expect(wrapper).toMatchSnapshot();
});

it('filters robots correctly', () => {
  expect(wrapper.instance().filterRobots()).toEqual([{
    id:3,
    name: 'John',
    email: 'john@gmail.com'
  } ]);
});


it('filters robots correctly with params', () => {
  const mockProps2 = {
    onRequestRobots:jest.fn(),
    robots:[{
      id:3,
      name: 'John',
      email: 'john@gmail.com'
    } ],
    searchField:'a',
    isPending: false
  }
  const filteredRobots = [];
  const wrapper2 = shallow(<MainPage {...mockProps2}/>)
  expect(wrapper2.instance().filterRobots([])).toEqual([]);
});


it('filters robots correctly 2', () => {
  const mockProps3 = {
    onRequestRobots:jest.fn(),
    robots:[{
      id:3,
      name: 'John',
      email: 'john@gmail.com'
    } ],
    searchField:'John',
    isPending: false
  }
  const filteredRobots = [];
  const wrapper3 = shallow(<MainPage {...mockProps3}/>)
  expect(wrapper3.instance().filterRobots()).toEqual([{
    id:3,
    name: 'John',
    email: 'john@gmail.com'
  }]);
});
