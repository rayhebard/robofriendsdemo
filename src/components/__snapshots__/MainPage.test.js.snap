// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    isPending={false}
    onRequestRobots={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    robots={
      Array [
        Object {
          "email": "john@gmail.com",
          "id": 3,
          "name": "John",
        },
      ]
    }
    searchField=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <SearchBox />,
        <Scroll>
          <ErrorBoundry>
            <CardList
              robots={
                Array [
                  Object {
                    "email": "john@gmail.com",
                    "id": 3,
                    "name": "John",
                  },
                ]
              }
            />
          </ErrorBoundry>
        </Scroll>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchChange": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ErrorBoundry>
            <CardList
              robots={
                Array [
                  Object {
                    "email": "john@gmail.com",
                    "id": 3,
                    "name": "John",
                  },
                ]
              }
            />
          </ErrorBoundry>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <CardList
              robots={
                Array [
                  Object {
                    "email": "john@gmail.com",
                    "id": 3,
                    "name": "John",
                  },
                ]
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "robots": Array [
                Object {
                  "email": "john@gmail.com",
                  "id": 3,
                  "name": "John",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <SearchBox />,
          <Scroll>
            <ErrorBoundry>
              <CardList
                robots={
                  Array [
                    Object {
                      "email": "john@gmail.com",
                      "id": 3,
                      "name": "John",
                    },
                  ]
                }
              />
            </ErrorBoundry>
          </Scroll>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchChange": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ErrorBoundry>
              <CardList
                robots={
                  Array [
                    Object {
                      "email": "john@gmail.com",
                      "id": 3,
                      "name": "John",
                    },
                  ]
                }
              />
            </ErrorBoundry>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <CardList
                robots={
                  Array [
                    Object {
                      "email": "john@gmail.com",
                      "id": 3,
                      "name": "John",
                    },
                  ]
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "robots": Array [
                  Object {
                    "email": "john@gmail.com",
                    "id": 3,
                    "name": "John",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
